src/Paramore.Brighter.Outbox.MySql/MySqlOutboxSync.cs

```diff
@@ -128,31 +137,38 @@ public void Add(Message message, int outBoxTimeout = -1, IAmABoxTransactionConne
             if (transactionConnectionProvider != null && transactionConnectionProvider is IMySqlTransactionConnectionProvider provider)
                 connectionProvider = provider;
 
-            var connection = connectionProvider.GetConnection();
-
-
-            if (connection.State != ConnectionState.Open) await connection.OpenAsync(cancellationToken).ConfigureAwait(ContinueOnCapturedContext);
-            using (var command = connection.CreateCommand())
+            using (var connection = connectionProvider.GetConnection())
```

Comment:
I think the question here (checking with @preardon ) is whether you can close this connection - because you may not own it. We ought to determine if you own the connection, and only close if it you own it

Reply:
Right so @honkuan86 if I check this file: https://github.com/BrighterCommand/Brighter/blob/master/src/Paramore.Brighter.Outbox.MsSql/MsSqlOutbox.cs I can see that approach, use a try...finally and check for ownership has been used. It looks like a fix from @preardon that did not roll into other *SQL outbox code. That is the code that I think you need. //cc @holytshirt